### PR TITLE
added clustering for sliding event

### PR DESCRIPTION
**Pull request by jelmerborst**

### CODE REVIEW

File: src/hsmm_mvpy/models.py [hunk 1]
```diff
@@ -868,26 +924,34 @@ def sliding_event(self, n_events=None, color='k', figsize=(12,3), verbose=True,
...
+        if cpus > 1:
+            if np.shape(magnitudes) == 2:
+                magnitudes = np.tile(magnitudes, (len(parameters), 1, 1))
+            with mp.Pool(processes=self.cpus) as pool:
+                estimates = pool.starmap(self.EM, 
+                    zip(itertools.repeat(1), magnitudes, parameters, 
+                        itertools.repeat(maximization), itertools.repeat(magnitudes_to_fix), 
+                        itertools.repeat(parameters_to_fix), itertools.repeat(max_iteration),
+                        itertools.repeat(tolerance)))
+        else:
```
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:942]: Somehow your commits contains an old code. This if/else function is no longer necessary, we can just keep the content of the if loop and remove the if

File: src/hsmm_mvpy/utils.py
Comment (GWeindel) [inline src/hsmm_mvpy/utils.py:1]: I think we need a separate module for the clustering as it seems relatively independent of the other functions in utils

**Review by GWeindel — COMMENTED** [review]
Everything looks good except the conflict on estimate_single_event(). For the addition in utils, I wonder whether we shouldn't create another module to avoid cluttering the utils module. As the code you added seems independent of the rest of the utils methods I think it is a better thing to do

File: src/hsmm_mvpy/models.py (hunk 1 above)
Comment (jelmerborst) [reply src/hsmm_mvpy/models.py:942]: Without the if/else it doesn't run because you get nested pools.

File: src/hsmm_mvpy/utils.py
Comment (jelmerborst) [reply src/hsmm_mvpy/utils.py:1]: Sure, I'll move it to a new file.

File: src/hsmm_mvpy/models.py (hunk 1 above)
Comment (GWeindel) [reply src/hsmm_mvpy/models.py:942]: Indeed, I forgot about that

**Comment by jelmerborst** [comment]
Moved clustering to a separate module,  can be merged.

**Review by GWeindel — APPROVED** [review]

**Pull request merged**